torchgeo/datasets/copernicus/cloud_s3.py

```diff
@@ -0,0 +1,178 @@
+        'Oa11_radiance',
+        'Oa12_radiance',
+        'Oa13_radiance',
+        'Oa14_radiance',
+        'Oa15_radiance',
+        'Oa16_radiance',
+        'Oa17_radiance',
+        'Oa18_radiance',
+        'Oa19_radiance',
+        'Oa20_radiance',
+        'Oa21_radiance',
+    )
+    rgb_bands = ('Oa08_radiance', 'Oa06_radiance', 'Oa04_radiance')
+    cmap = ListedColormap(['red', 'gray', 'lightgray', 'white', 'black', 'snow'])
+    classes: tuple[str, ...] = (
+        'Invalid',
+        'Clear',
+        'Cloud-Ambiguous',
+        'Cloud-Sure',
+        'Cloud Shadow',
+        'Snow/Ice',
+    )
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: Literal['train', 'val', 'test'] = 'train',
+        mode: Literal['binary', 'multi'] = 'multi',
+        bands: Sequence[str] | None = None,
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new CopernicusBenchBase instance.
+
+        Args:
+            root: Root directory where dataset can be found.
+            split: One of 'train', 'val', or 'test'.
+            mode: One of 'binary' or 'multi'.
+            bands: Sequence of band names to load (defaults to all bands).
+            transforms: A function/transform that takes input sample and its target as
+                entry and returns a transformed version.
+            download: If True, download dataset and store it in the root directory.
+            checksum: If True, check the MD5 of the downloaded files (may be slow).
+
+        Raises:
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        self.mode = mode
+        if mode == 'binary':
+            self.classes = ('Invalid', 'Clear', 'Cloud')
+            self.cmap = ListedColormap(['red', 'gray', 'white'])
+        super().__init__(root, split, bands, transforms, download, checksum)
+
+    def _load_image(self, index: int) -> dict[str, Tensor]:
+        """Load an image.
+
+        Args:
+            index: Index to return.
+
+        Returns:
+            An image sample.
+        """
+        sample: dict[str, Tensor] = {}
+        file = self.files[index]
+        with rio.open(os.path.join(self.root, self.directory, 's3_olci', file)) as f:
+            sample['image'] = torch.tensor(f.read(self.band_indices).astype(np.float32))
+
+        return sample
+
+    def _load_target(self, index: int) -> dict[str, Tensor]:
+        """Load a target mask.
+
+        Args:
+            index: Index to return.
+
+        Returns:
+            A target sample.
+        """
+        sample: dict[str, Tensor] = {}
+        file = self.files[index]
+        mode = f'cloud_{self.mode}'
+        with rio.open(os.path.join(self.root, self.directory, mode, file)) as f:
+            sample['mask'] = torch.tensor(f.read(1).astype(np.int64))
```

Comment:
```suggestion
            sample['mask'] = torch.from_numpy(f.read(1)).long()
```

Reply:
This is actually what I used originally but it doesn't work for older PyTorch (would have to check now that we bumped the minimum PyTorch version). See https://github.com/microsoft/torchgeo/pull/2604/commits/399d0ae004a083fe89acd2d48ea849d7257d5dff for the change and https://github.com/microsoft/torchgeo/actions/runs/13882120290/job/38842005656 for the failing tests. Basically has to do with PyTorch support for uint16.